packages/nextjs/src/server/utils/withClerkMiddleware.ts

```diff
@@ -0,0 +1,132 @@
+import { AuthStatus } from '@clerk/backend-core';
+import { NextMiddleware, NextMiddlewareResult } from 'next/dist/server/web/types';
+import { NextFetchEvent, NextRequest, NextResponse } from 'next/server';
+
+import { vercelEdgeBase } from '../../edge-middleware';
+import {
+  AUTH_RESULT,
+  AuthResult,
+  NEXT_REDIRECT_HEADER,
+  NEXT_RESUME_HEADER,
+  NEXT_REWRITE_HEADER,
+  SESSION_COOKIE_NAME,
+} from '../types';
+import { getAuthResultFromRequest, setAuthResultOnRequest } from './requestResponseUtils';
+
+const DEFAULT_API_URL = process.env.CLERK_API_URL || 'https://api.clerk.dev';
+const DEFAULT_API_VERSION = process.env.CLERK_API_VERSION || 'v1';
+
+// Using public interstitial endpoint for time being
+const INTERSTITIAL_URL = `${DEFAULT_API_URL}/${DEFAULT_API_VERSION}/public/interstitial?frontendApi=${process.env.NEXT_PUBLIC_FRONTEND_API}`;
```

Comment:
❓ Should this be `NEXT_PUBLIC_CLERK_FRONTEND_API` ?

Reply:
Fixed